### ISSUE TITLE
Upload progress is actually progress of creating a temp file?

### ISSUE THREAD

**Issue by ParkerK** [kittinunf/Fuel#193]
In [UploadTaskRequest.kt](https://github.com/kittinunf/Fuel/blob/master/fuel/src/main/kotlin/com/github/kittinunf/fuel/core/requests/UploadTaskRequest.kt), the progress callback isn't sending the progress of the file upload, it's incorrectly sending the progress of writing the file to the output stream

Line 42:
```
    //input file data
    if (outputStream != null) {
       FileInputStream(file).use {
           it.copyTo(outputStream, BUFFER_SIZE) { writtenBytes ->
               progressCallback?.invoke(contentLength + writtenBytes, totalLength)
             }
         }
    }
  contentLength += file.length()
  contentLength += write(CRLF)
```

So does that mean when progress is 100%,  that the upload has been submitted to the server, but it doesn't mean that the upload has completed uploading to the remote server?

**Comment by Gira-X** [comment]
There is an old Github Gist git diff (from May 2017, not sure if this can be still applied to the current Fuel version) which fixes this issue partly: https://gist.github.com/dstd/cc5710c52c58eec6f14ee1eb229eb4de

The applied fix only works for `Fuel.download`.
For `Fuel.upload` it still reports the wrong progress.

Maybe someone can take a look and completely fix this :)

**Comment by SleeplessByte** [comment]
Fixed with #485
